### PR TITLE
Dead letter queue

### PR DESCRIPTION
**Pull request by mkrzemien**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,82 +1,91 @@
...
+version := "0.3.0-SNAPSHOT"
 
-lazy val commonSettings = Seq(
-  organization := "com.softwaremill.kmq",
-  version := "0.2.3",
-  scalaVersion := "2.12.4",
-  crossScalaVersions := List(scalaVersion.value, "2.11.11"),
+val scala2_12 = "2.12.16"
+val scala2_13 = "2.13.8"
+val scala3 = "3.1.2"
```
Comment (adamw) [inline build.sbt:12]: This is unused I think?

File: build.sbt [hunk 2]
```diff
@@ -1,82 +1,91 @@
+import com.softwaremill.Publish.ossPublishSettings
```
Comment (adamw) [inline build.sbt:1]: would also be good to add CI + publishing as in our other projets, see e.g. sttp-model for a simple example on how to configure github actions

File: core/src/main/java/com/softwaremill/kmq/KmqConfig.java [hunk 3]
```diff
@@ -4,32 +4,55 @@
  * Configuration for the Kafka-based MQ.
  */
 public class KmqConfig {
+    private static final int MAX_REDELIVERY_COUNT = 3;
+    private static final String REDELIVERY_COUNT_HEADER = "kmq-redelivery-count";
+    private static final String DEAD_LETTER_TOPIC_SUFFIX = "__undelivered";
```
Comment (adamw) [inline core/src/main/java/com/softwaremill/kmq/KmqConfig.java:9]: [minor] these are defaults, so maybe add `DEFAULT_` suffix/prefix?

File: example-java/src/main/java/com/softwaremill/kmq/example/standalone/StandaloneConfig.java [hunk 4]
```diff
@@ -6,10 +6,10 @@
 import java.time.Duration;
 
 class StandaloneConfig {
-    static final KmqConfig KMQ_CONFIG = new KmqConfig("queue", "markers", "kmq_client",
-            "kmq_redelivery", Duration.ofSeconds(90).toMillis(), 1000);
+    static final KmqConfig KMQ_CONFIG = new KmqConfig("queue", "markers", "kmq_client", "kmq_redelivery",
+            Duration.ofSeconds(90).toMillis(), 1000);
 
-    /* EXAMPLE with extraConfig : SSL Encryption & SSL Authentication
+    /* TODO: EXAMPLE with extraConfig : SSL Encryption & SSL Authentication
```
Comment (adamw) [inline example-java/src/main/java/com/softwaremill/kmq/example/standalone/StandaloneConfig.java:12]: is there something to do here?

**Review by adamw — APPROVED** [review]
looks good, let's add CI and merge :)

File: example-java/src/main/java/com/softwaremill/kmq/example/standalone/StandaloneConfig.java (hunk 4 above)
Comment (mkrzemien) [reply example-java/src/main/java/com/softwaremill/kmq/example/standalone/StandaloneConfig.java:12]: No, I was just my first reaction when I saw commented-out code :) 
On the second thought it's okay in an example.

File: build.sbt (hunk 1 above)
Comment (mkrzemien) [reply build.sbt]: Okay, I'm removing scala3 for now.

File: build.sbt (hunk 2 above)
Comment (mkrzemien) [reply build.sbt:1]: Okay

**Pull request merged**